service_common.go

```diff
@@ -21,23 +21,23 @@ func (c *controller) addEndpointNameResolution(svcName, svcID, nID, eID, contain
 	c.addContainerNameResolution(nID, eID, containerName, taskAliases, ip, method)
 
 	// Add endpoint IP to special "tasks.svc_name" so that the applications have access to DNS RR.
-	n.(*network).addSvcRecords(eID, "tasks."+svcName, ip, nil, false, method)
+	n.(*network).addSvcRecords(eID, "tasks."+svcName, svcID, ip, nil, false, method)
 	for _, alias := range serviceAliases {
-		n.(*network).addSvcRecords(eID, "tasks."+alias, ip, nil, false, method)
+		n.(*network).addSvcRecords(eID, "tasks."+alias, svcID, ip, nil, false, method)
 	}
 
 	// Add service name to vip in DNS, if vip is valid. Otherwise resort to DNS RR
 	if len(vip) == 0 {
-		n.(*network).addSvcRecords(eID, svcName, ip, nil, false, method)
+		n.(*network).addSvcRecords(eID, svcName, eID, ip, nil, false, method)
```

Comment:
Why is DNS-RR treated as an attachable container use-case, while the svcID can still be valid.